File: justcall_server/socket/socket.js
import { Server } from "socket.io";
import eventType from "../type/enum.socketEvent.mjs";
import controllers from "./controllers/controllers.js";
import SocketData from "./storage.js";
let IO;
const channels = SocketData.channels;
function socketIoInit(server) {
  IO = new Server(server, {
    cors: {
      origin: "*",
      methods: ["GET", "POST"],
    },
  });
  IO.use((socket, next) => {
    if (socket.handshake.query) {
      SocketData.usersSocket[socket.handshake.query._id] = {
        socket,
      };
      socket.user_id = socket.handshake.query._id;
    }
    next();
  });
  IO.on(eventType.connection, (socket) => {
    SocketData.usersSocket[socket.user_id].socket = socket;
    socket.on(eventType.disconnect, () => {
      delete SocketData.usersSocket[socket.user_id];
    });
    socket.on(eventType.singleCall, (data) =>
      controllers.singleCall(data, socket)
    );
    socket.on(eventType.singleAns, (data) =>
      controllers.singleAns(data, socket)
    );
    socket.on(eventType.singleIceCandidate, (data) =>
      controllers.singleIceCandidate(data, socket)
    );
    socket.on(eventType.groupJoin, (data) =>
      controllers.groupJoin(data, socket)
    );
    socket.on(eventType.groupSDSet, (data) =>
      controllers.groupSDSet(data, socket)
    );
    socket.on(eventType.groupSDComplite, (data) =>
      controllers.groupSDComplite(data, socket)
    );
    socket.on(eventType.groupIceCandidate, (data) =>
      controllers.groupIceCandidate(data, socket)
    );
    socket.on(eventType.conferenceJoin, (data) =>
      controllers.conferenceJoin(data, socket)
    );
    socket.on(eventType.conferenceSDSet, (data) =>
      controllers.conferenceSDSet(data, socket)
    );
    socket.on(eventType.conferenceSDComplite, (data) =>
      controllers.conferenceSDComplite(data, socket)
    );
    socket.on(eventType.conferenceIceCandidate, (data) =>
      controllers.conferenceIceCandidate(data, socket)
    );
    socket.on(eventType.outOfApplication, (data) =>
      controllers.outOfApplication(data, socket)
    );
    socket.on(eventType.returnOnApplication, (data) =>
      controllers.returnOnApplication(data, socket)
    );
  });
}

export default { IO, socketIoInit };

File: justcall/lib/network/socket.dart
import 'package:justcall/network/network.dart';

class Socket {
  final _socket = InitForUserData.getSocket();
  void socketInit() {
    _socket.on(SocketEvent.singleCall, (data) {});
    _socket.on(SocketEvent.singleAns, (data) {});
    _socket.on(SocketEvent.singleIceCandidate, (data) {});
    _socket.on(SocketEvent.groupJoin, (data) {});
    _socket.on(SocketEvent.groupSDSet, (data) {});
    _socket.on(SocketEvent.groupSDComplite, (data) {});
    _socket.on(SocketEvent.groupIceCandidate, (data) {});
    _socket.on(SocketEvent.conferenceJoin, (data) {});
    _socket.on(SocketEvent.conferenceSDSet, (data) {});
    _socket.on(SocketEvent.conferenceSDComplite, (data) {});
    _socket.on(SocketEvent.conferenceIceCandidate, (data) {});
  }
}

class SocketEvent {
  static const String singleCall = 'singleCall';
  static const String singleAns = 'singleAns';
  static const String singleIceCandidate = 'singleIceCandidate';
  static const String groupJoin = 'groupJoin';
  static const String groupSDSet = 'groupSDSet';
  static const String groupSDComplite = 'groupSDComplite';
  static const String groupIceCandidate = 'groupIceCandidate';
  static const String conferenceJoin = 'conferenceJoin';
  static const String conferenceSDSet = 'conferenceSDSet';
  static const String conferenceSDComplite = 'conferenceSDComplite';
  static const String conferenceIceCandidate = 'conferenceIceCandidate';
}
